neurom/io/utils.py

```diff
@@ -108,14 +107,51 @@ def get_files_by_path(path):
     raise IOError('Invalid data path %s' % path)
 
 
-def load_neuron(handle, reader=None):
-    """Build section trees from an h5 or swc file."""
-    if isinstance(handle, str):
-        handle = Path(handle)
+def load_neuron(neuron, reader=None):
+    """Build section trees from a neuron or a h5, swc or asc file.
 
-    rdw = load_data(handle, reader)
-    name = handle.stem if isinstance(handle, Path) else None
-    return FstNeuron(rdw, name)
+    Args:
+        neuron (str|Path|Neuron|morphio.Morphology|morphio.mut.Morphology): A neuron representation
+            It can be:
+            - a filename with the h5, swc or asc extension
+            - a NeuroM Neuron object
+            - a morphio mutable or immutable Morphology object
+            - a stream that can be put into a io.StreamIO object. In this case, the READER argument
+            must be passed with the corresponding file format (asc, swc and h5)
+        reader (str): Optional, must be provided if neuron is a stream to
+                      specify the file format (asc, swc, h5)
+
+    Returns:
+        A Neuron object
+
+    Examples:
+            neuron = neurom.load_neuron('my_neuron_file.h5')
+
+            neuron = neurom.load_neuron(morphio.Morphology('my_neuron_file.h5'))
+
+            neuron = nm.load_neuron(io.StringIO('''
+            ((Dendrite)
+            (3 -4 0 2)
+            (3 -6 0 2)
+            (3 -8 0 2)
+            (3 -10 0 2)
+            (
+            (0 -10 0 2)
+            (-3 -10 0 2)
+            |
+            (6 -10 0 2)
+            (9 -10 0 2)
+            )
+            )'''), reader='asc')
+    """
+    if isinstance(neuron, (Neuron, morphio.Morphology, morphio.mut.Morphology)):
+        return Neuron(neuron)
+
+    if reader:
+        return Neuron(_get_file(neuron, reader))
+
+    name = os.path.splitext(os.path.basename(neuron))[0]
```

Comment:
we don't use pathlib in this repo, right?

Reply:
I didn't fix those minors, to ease rebases of this branch. There are plenty of minors like this in the project which I didn't touch.